### ISSUE TITLE
bookie_journal_JOURNAL_QUEUE_SIZE Prometheus metric is negative (Pulsar 2.6.1)

### ISSUE THREAD

**Issue by frankjkelly** [apache/pulsar#9634]
**Describe the bug**
The prometheus metric `bookie_journal_JOURNAL_QUEUE_SIZE` exposed by bookie is negative. Shouldn't queue lengths be positive

**To Reproduce**
1. Launch Pulsar on Kubernetes
2. Inspect metrics endpoint of Bookkeeper
```
$ curl http://platform-pulsar-bookkeeper:8000/metrics | grep QUEUE_SIZE
# TYPE bookie_journal_JOURNAL_CB_QUEUE_SIZE counter
bookie_journal_JOURNAL_CB_QUEUE_SIZE 57253790
# TYPE bookie_journal_JOURNAL_FORCE_WRITE_QUEUE_SIZE counter
bookie_journal_JOURNAL_FORCE_WRITE_QUEUE_SIZE 0
# TYPE bookie_journal_JOURNAL_QUEUE_SIZE counter
bookie_journal_JOURNAL_QUEUE_SIZE -57253790
```

**Expected behavior**
Queue lengths are zero or above

**Screenshots**
![image](https://user-images.githubusercontent.com/62910985/108514728-34b4fc00-7292-11eb-9038-ad435fc9f12c.png)

**Additional context**
- Pulsar `2.6.1`
- Kafkaesque Helm Chart `1.0.24`
- Probably has not bearing but `exposeTopicLevelMetricsInPrometheus` is set to `false` as we have a lot of topics

**Comment by michaeljmarshall** [comment]
I started looking into this tonight. I believe this issue belongs in the [apache/bookkeeper](https://github.com/apache/bookkeeper) repo because the metric is defined there.

Regarding the problematic metric, I already see some irregularities based on parameter names and things getting passed around.

@frankjkelly can you confirm that your only seeing this negative value for `bookie_journal_JOURNAL_QUEUE_SIZE`?

**Comment by michaeljmarshall** [comment]
I found some interesting things that I wanted to share, so I opened a copy of this issue in bookkeeper. I think we should continue all work against this issue there.

**Comment by frankjkelly** [comment]
@michaeljmarshall Yes only `bookie_journal_JOURNAL_QUEUE_SIZE` is negative - see below where `bookie_journal_JOURNAL_CB_QUEUE_SIZE` is positive (and almost the exact inverse value)
![image](https://user-images.githubusercontent.com/62910985/108849681-730f2b80-75b0-11eb-90de-0b30f646f330.png)

**Comment by michaeljmarshall** [comment]
> bookie_journal_JOURNAL_CB_QUEUE_SIZE is positive (and almost the exact inverse value)

That's a helpful observation. When I was looking through the code last night, I was wondering if there was any type of relationship between those two metrics.

**Comment by michaeljmarshall** [comment]
@frankjkelly - I believe you're running with apache bookkeeper `4.10.0` (that is what shipped in the pulsar docker image for pulsar `2.6.1`. This bug was fixed for bookkeeper `4.10.1`, as seen in this PR: https://github.com/apache/bookkeeper/pull/2290. Pulsar `2.6.2` is `4.10.0` too, but Pulsar `2.6.3` comes with bookkeeper `4.11.1`, which has the fix to the metric.

**Comment by frankjkelly** [comment]
@michaeljmarshall Thanks so much for hunting this down. Closing this ticket . . . .
